app/containers/App.js

```diff
@@ -0,0 +1,221 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2014-2016, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+import _ from 'lodash';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { remote } from 'electron';
+
+const {Menu} = remote;
+
+import bows from '../../lib/bows.js';
+
+import config from '../../lib/config.js';
+
+import carelink from '../../lib/core/carelink.js';
+import device from '../../lib/core/device.js';
+import localStore from '../../lib/core/localStore.js';
+
+import actions from '../actions/';
+const asyncActions = actions.async;
+const syncActions = actions.sync;
+
+import * as actionSources from '../constants/actionSources';
+import { pages, urls, pagesMap } from '../constants/otherConstants';
+import { checkVersion } from '../utils/drivers';
+
+import UpdatePlease from '../components/UpdatePlease';
+import VersionCheckError from '../components/VersionCheckError';
+import Footer from '../components/Footer';
+import Header from '../components/Header';
+import UpdateModal from '../components/UpdateModal';
+
+import styles from '../../styles/components/App.module.less';
+
+export class App extends Component {
+  static propTypes = {
+    // api: PropTypes.func.isRequired
```

Comment:
I'm actually going to re-work this because the `api` moved into the `route` prop, but is still absolutely necessary for everything else to work.